cmd/tendermint/commands/init.go

```diff
@@ -86,7 +87,11 @@ func initFilesWithConfig(config *cfg.Config) error {
 				PubKeyTypes: []string{types.ABCIPubKeyTypeSecp256k1},
 			}
 		}
-		pubKey, err := pv.GetPubKey()
+
+		ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+		defer cancel()
```

Comment:
I think eventually there should probably be one `context.Context` object per-process invocation and that all contexts should be descended from (to support being able to cancel all of the contexts from a signal handler, or some such,) rather than making these one off contexts.

Having said that, (of course), this is really fine and I don't have any functional changes to this patch, except that I'd change `context.Background()` to `context.TODO()` to indicate an intention to eventually move toward a single context tree.

Reply:
Changed the backgrounds, not in tests, to `TODO()`. Thanks for the pointer